WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -19,7 +20,42 @@ def new
   end
 
   def index
-    @competitions = Competition.all.select([:id, :name, :cityName, :countryId]).order(:year, :month, :day).reverse_order
+    @regions = { '' => [["All","all"]],
+                 'Continent' => Continent.all.map { |continent| [continent.name, continent.id] },
+                 'Country' => Country.all.map { |country| [country.name, country.id] } }
+    @events = [ ["All", "all"], ["",""] ] + Event.all_official.map { |event| [event.name, event.id] }
+    @years = [ ["Current","current"],["All","all"],["",""] ] + Competition.select(:year).map(&:year).uniq.reverse!
```

Comment:
Nitpick: I'm currently seeing the year 0 on https://staging.worldcubeassociation.org/competitions. Is that because you're querying *all* competitions, not just ones where `showAtAll` is true?